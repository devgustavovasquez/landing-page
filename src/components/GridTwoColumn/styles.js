import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.spaces.large};

    @media ${theme.medias.lteMedium} {
      grid-template-columns: 1fr;
      text-align: center;
    }

    ${Title} {
      margin-bottom: ${theme.spaces.xlarge};
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    @media ${theme.medias.lteMedium} {
      margin-bottom: ${theme.spaces.large};
    }
  `}
`;

export const ImageContainer = styled.div`
  ${({ theme }) => css``}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 100%;
  `}
`;
