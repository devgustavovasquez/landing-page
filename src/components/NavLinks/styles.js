import styled, { css } from 'styled-components';

export const Container = styled.nav`
  ${({ theme }) => css`
    display: flex;
    flex-flow: row wrap;

    @media ${theme.medias.lteMedium} {
      flex-flow: column wrap;
      align-items: center;
    }
  `}
`;
