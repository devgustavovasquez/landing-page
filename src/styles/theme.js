export const theme = {
  colors: {
    mainBg: 'pink',
    secondaryBg: 'blue',
  },
  fonts: {},
  spaces: {},
};
