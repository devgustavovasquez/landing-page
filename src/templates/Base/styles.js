import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    padding-top: 5.4rem;

    @media ${theme.medias.lteMedium} {
      padding-top: 0;
    }
  `}
`;
