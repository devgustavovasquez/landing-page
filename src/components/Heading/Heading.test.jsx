import { screen } from '@testing-library/react';
import { Heading } from '.';
import { renderTheme } from '../../styles/render-theme';
import { theme } from '../../styles/theme';

describe('<Heading />', () => {
  test('should render with default values', () => {
    renderTheme(<Heading>texto</Heading>);

    const heading = screen.getByRole('heading', { name: 'texto' });
    expect(heading).toHaveStyle({
      color: theme.colors.primaryColor,
      'font-size': theme.fonts.sizes.xhuge,
      'text-transform': 'none',
    });
  });

  test('should render with white color', () => {
    renderTheme(<Heading colorDark={false}>texto</Heading>);

    const heading = screen.getByRole('heading', { name: 'texto' });
    expect(heading).toHaveStyle({
      color: theme.colors.white,
    });
  });

  test('should render correct small size', () => {
    renderTheme(<Heading size="small">texto</Heading>);

    const heading = screen.getByRole('heading', { name: 'texto' });
    expect(heading).toHaveStyle({
      'font-size': theme.fonts.sizes.medium,
    });
  });

  test('should render correct medium size', () => {
    renderTheme(<Heading size="medium">texto</Heading>);

    const heading = screen.getByRole('heading', { name: 'texto' });
    expect(heading).toHaveStyle({
      'font-size': theme.fonts.sizes.large,
    });
  });

  test('should render correct big size', () => {
    renderTheme(<Heading size="big">texto</Heading>);

    const heading = screen.getByRole('heading', { name: 'texto' });
    expect(heading).toHaveStyle({
      'font-size': theme.fonts.sizes.xlarge,
    });
  });

  test('should render correct font size in mobile', () => {
    renderTheme(<Heading size="huge">texto</Heading>);

    const heading = screen.getByRole('heading', { name: 'texto' });
    expect(heading).toHaveStyleRule('font-size', theme.fonts.sizes.xlarge, { media: theme.medias.lteMedium });
  });

  test('should render with uppercase letters', () => {
    renderTheme(<Heading uppercase>texto</Heading>);

    const heading = screen.getByRole('heading', { name: 'texto' });
    expect(heading).toHaveStyle({
      'text-transform': 'uppercase',
    });
  });

  test('should match snapshot', () => {
    const { container } = renderTheme(<Heading uppercase>Children</Heading>);
    expect(container).toMatchSnapshot();
  });
});
