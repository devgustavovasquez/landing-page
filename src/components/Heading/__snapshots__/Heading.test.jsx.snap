// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should match snapshot 1`] = `
.c0 {
  color: #0a1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

@media (max-width:768px) {
  .c0 {
    font-size: 4.0rem;
  }
}

<div>
  <h1
    class="c0"
  >
    Children
  </h1>
</div>
`;
